### DISCUSSION TITLE
MQTT rx missing on FW v1.71 and HW v3.00_obd

### DISCUSSION THREAD

**Discussion by cmangla** [meatpiHQ/wican-fw#45] [Q&A]
Since updating my WiCAN to firmware v1.71, it is a lot more stable, and it connects to my MQTT broker with `status=online`, but there is no `rx` topic. I tried the example node-red flow, put in my device IDs, but it doesn't seem to receive any MQTT messages other than the status. I see it sending to the `tx` topic.

Before updating I could see a continuous stream of messages on the `rx` topic.

I did an OTA update, which took a few tries.

Any advise on a fix?

**Comment by meatpiHQ** [comment]
@cmangla MQTT rx should still work in v1.71. Can you please share a screenshot of your configuration webpage?

**Reply by cmangla** [reply]
@meatpiHQ Here you go:

![image](https://github.com/meatpiHQ/wican-fw/assets/7121280/fe211cac-43ce-4bac-9c90-d3b53043a659)
![image](https://github.com/meatpiHQ/wican-fw/assets/7121280/d4ae51af-acf0-4a1f-b2d4-19bfa460f340)

**Reply by cmangla** [reply]
@meatpiHQ Some more screenshots, just in case:

![image](https://github.com/meatpiHQ/wican-fw/assets/7121280/8bf9ddb7-ac87-4db2-9d6c-823cb0fc564d)
![image](https://github.com/meatpiHQ/wican-fw/assets/7121280/8e0071a1-c935-4ab6-a14d-5877ceb109b0)

**Comment by meatpiHQ** [comment]
@cmangla Please try to disable battery alerts and try again.

**Reply by cmangla** [reply]
Just tried that and I see no improvement unfortunately. On reboot, wican sends status message, but then there is nothing else on MQTT. I'm checking using MQTT Explorer.

**Comment by meatpiHQ** [comment]
@cmangla It works for me. 

Try to subscribe to this topic "wican/+/can/rx" see if anything comes in. 

Is the orange LED blinking? if it's not that means WiCAN doesn't see any CAN traffic. 

Did you use the monitor page and did not reboot afterwards?

**Reply by cmangla** [reply]
@meatpiHQ 

I managed to get it working by starting up my car.

> Is the orange LED blinking? if it's not that means WiCAN doesn't see any CAN traffic.

I noticed the orange LED wasn't blinking after my car had been parked for a while. So I started up my car, and it started blinking, and data is coming through the rx channel on mqtt. My car is now turned-off but I can still see data coming through on MQTT.

**Reply by cmangla** [reply]
The rx channel has gone silent now just as I finished typing this message.

**Reply by cmangla** [reply]
Also, wican is not in sleep mode. The battery is at 12.7v and sleep threshold is 12.5v.

**Comment by meatpiHQ** [comment]
@cmangla 

> The rx channel has gone silent now just as I finished typing this message.

When you turn off the car the ECU goes to sleep.

> Also, wican is not in sleep mode. The battery is at 12.7v and sleep threshold is 12.5v.

WiCAN will only go the sleep if the voltage reading goes under the 12.7V threshold you've set. And there is also a delay before it goes to sleep, I suggest to set the threshold around 13V and wait sometime. The blue LED will start blinking every 1 second to it's in sleep mode.
